cocotb/handle.py

```diff
@@ -486,7 +475,7 @@ def __setitem__(self, index, value):
             if len(value) != len(self.__getitem__(index)):
                 raise IndexError("Assigning list of length %d to object %s of length %d" % (
                     len(value), self.__getitem__(index)._fullname, len(self.__getitem__(index))))
-            self._log.info("Setting item %s to %s", self.__getitem__(index)._fullname, value)
+            self._log.debug("Setting item %s to %s", self.__getitem__(index)._fullname, value)
             for idx in range(len(value)):
                 self.__getitem__(index).__setitem__(idx, value[idx])
         else:
```

Comment:
Can these `__getitem__` calls be changed to `[]` syntax?

Reply:
Yes. Better yet, this entire `if` statement can now be replaced with it's else clause, I think.

Reply:
If you want to go down this road, please add a test for 2d arrays.

Reply:
I'm glad I checked, because...
```python
for idx in range(len(value)):
```
...is wrong.

Reply:
Well spotted!